mappers/kubeedge-v1.17.0/mqtt-mapper/driver/devicetype.go

```diff
@@ -0,0 +1,78 @@
+package driver
+
+import (
+	"sync"
+	mqtt "github.com/eclipse/paho.mqtt.golang"
+	"github.com/kubeedge/mapper-framework/pkg/common"
+)
+
+// CustomizedDev is the customized device configuration and client information.
+type CustomizedDev struct {
+	CustomizedClient *CustomizedClient
+	Instance         common.DeviceInstance
+}
+
+type CustomizedClient struct {
+	// TODO add some variables to help you better implement device drivers
+	deviceMutex    sync.Mutex
+	ProtocolConfig
+	MessageQueue MessageQueue
+}
+
+type ProtocolConfig struct {
+	ProtocolName string   `json:"protocolName"`
+	ConfigData            `json:"configData"`
+}
+
+type ConfigData struct {
+	// MQTT protocol config data
+    ClientID     string `json:"clientID"`  // MQTT Client ID
+    Topic        string `json:"topic"`      // Client need to specify a topic when publishing or subsribing.
+    Message      string `json:"message"`    // Content of the message
+}
+
+
+type VisitorConfig struct {
+	ProtocolName      string  `json:"protocolName"`
+	VisitorConfigData         `json:"configData"`
+}
+
+type VisitorConfigData struct {
+	DataType string `json:"dataType"`
+
+	ClientID          string                 `json:"clientID"`      // MQTT Client ID
+	DeviceInfo        string                 `json:"deviceInfo"`    // Device information, such as device identification or other important information.
+	OperationInfo     OperationInfoType      `json:"operationInfo"` // Operation information, such as adding, deleting, modifying and so on.
+	SerializedFormat  SerializedFormatType   `json:"fileType"`      // Supported formats: json, xml and yaml.
+	ParsedMessage     map[string]interface{} `json:"parsedMessage"` // The parsed message
```

Comment:
Because mapper is a device management plug-in in kubeedge, it needs to complete the function of communicating with the actual edge IoT device and obtaining device data. According to your definition here, can mapper obtain the information and data of the actual mqtt device by subscribing to this ClientID?

Reply:
Thank you for your feedback! I have enriched the MQTT configuration fields in ConfigData.